Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -1191,23 +1225,24 @@ the variables, given by `m ↦ f (m 0) (tail m)`-/
 def LinearMap.uncurryLeft (f : M 0 →ₗ[R] MultilinearMap R (fun i : Fin n => M i.succ) M₂) :
     MultilinearMap R M M₂ where
   toFun m := f (m 0) (tail m)
-  map_add' dec m i x y := by
-    rw [Subsingleton.elim dec (by infer_instance)]
+  map_add' := @fun dec m i x y => by
+    -- porting note: `clear` not necessary in Lean 3 due to not being in the instance cache
+    rw [Subsingleton.elim dec (by clear dec; infer_instance)]; clear dec
     by_cases h : i = 0
     · subst i
-      rw [update_same, update_same, update_same, f.map_add, add_apply, tail_update_zero,
-        tail_update_zero, tail_update_zero]
-    · rw [update_noteq (Ne.symm h), update_noteq (Ne.symm h), update_noteq (Ne.symm h)]
+      simp only [update_same, map_add, tail_update_zero, MultilinearMap.add_apply]
+    · simp_rw [update_noteq (Ne.symm h)]
       revert x y
       rw [← succ_pred i h]
       intro x y
       rw [tail_update_succ, MultilinearMap.map_add, tail_update_succ, tail_update_succ]
-  map_smul' dec m i c x := by
-    rw [Subsingleton.elim dec (by infer_instance)]
+  map_smul' := @fun dec m i c x => by
+    -- porting note: `clear` not necessary in Lean 3 due to not being in the instance cache
+    rw [Subsingleton.elim dec (by clear dec; infer_instance)]; clear dec
     by_cases h : i = 0
     · subst i
-      rw [update_same, update_same, tail_update_zero, tail_update_zero, ← smul_apply, f.map_smul]
-    · rw [update_noteq (Ne.symm h), update_noteq (Ne.symm h)]
+      simp only [update_same, map_smul, tail_update_zero, MultilinearMap.smul_apply]
+    · simp_rw [update_noteq (Ne.symm h)]
```

Comment:
This change might not be necessary after all

Reply:
The change is necessary – it doesn't compile otherwise.